docs/extensions/controllerregistration.md

```diff
@@ -201,12 +201,30 @@ spec:
     primary: true
     globallyEnabled: true
     reconcileTimeout: 30s
+    lifecycle:
+      reconcile: AfterKubeAPIServer
+      delete: BeforeKubeAPIServer
+      migrate: BeforeKubeAPIServer
 ```
 
 The `globallyEnabled=true` option specifies that the `Extension/foo` object shall be created by default for all shoots (unless they opted out by setting `.spec.extensions[].enabled=false` in the `Shoot` spec).
 
 The `reconcileTimeout` tells Gardener how long it should wait during its shoot reconciliation flow for the `Extension/foo`'s reconciliation to finish.
 
+#### `Extension` lifecycle
```

Comment:
Probably add a link to this section from https://github.com/gardener/gardener/blob/master/docs/extensions/extension.md